Source/Core/VideoBackends/OGL/OGLTexture.cpp

```diff
@@ -164,15 +149,73 @@ void OGLTexture::Bind(unsigned int stage)
   }
 }
 
-bool OGLTexture::Save(const std::string& filename, unsigned int level)
+std::optional<AbstractTexture::RawTextureInfo> OGLTexture::MapFullImpl()
+{
+  if (GLInterface->GetMode() != GLInterfaceMode::MODE_OPENGL)
+    return {};
+
+  m_staging_data.reserve(m_config.width * m_config.height * 4);
+  glActiveTexture(GL_TEXTURE9);
+
+  glBindTexture(GL_TEXTURE_2D_ARRAY, m_texId);
+  glGetTexImage(GL_TEXTURE_2D_ARRAY, 0, GL_RGBA, GL_UNSIGNED_BYTE, m_staging_data.data());
+  OGLTexture::SetStage();
+  return AbstractTexture::RawTextureInfo{reinterpret_cast<u8*>(m_staging_data.data()),
+                                         m_config.width * 4, m_config.width, m_config.height};
+}
+
+std::optional<AbstractTexture::RawTextureInfo> OGLTexture::MapRegionImpl(u32 level, u32 x, u32 y,
+                                                                         u32 width, u32 height)
 {
-  // We can't dump compressed textures currently (it would mean drawing them to a RGBA8
-  // framebuffer, and saving that). TextureCache does not call Save for custom textures
-  // anyway, so this is fine for now.
-  _assert_(m_config.format == AbstractTextureFormat::RGBA8);
+  if (GLInterface->GetMode() != GLInterfaceMode::MODE_OPENGL)
+    return {};
+  m_staging_data.reserve(m_config.width * m_config.height * 4);
+  glActiveTexture(GL_TEXTURE9);
+  glBindTexture(GL_TEXTURE_2D_ARRAY, m_texId);
+  if (GLExtensions::Supports("ARB_get_texture_sub_image"))
```

Comment:
Probably better to cache this in a variable somewhere, rather than doing a string/hash table lookup every time. But probably not a huge impact in the overall picture.

Reply:
Good point.  Thinking I might just put it as a static on the texture for now?

Reply:
It's not per-texture state. Renderer is probably the most appropriate place for it, IMO.

Reply:
Sounds good, I'll do that.  Thanks.

Reply:
Sorry, I forgot g_ogl_config was a thing, and would be a better place. That's where the rest of the feature bits are set.